### PR TITLE
Change DepositStatusPill to use Chip component

### LINKED ISSUE #6818: Deposits status chip should use the `Chip` component
As part of #6794, we are updating the deposit status "chip" elements to use the `Chip` component rather than the `Pill` component.

https://github.com/Automattic/woocommerce-payments/blob/7e685b9e4355fc2e4c403ea9388c3d0dcc685a7f/client/components/deposit-status-pill/index.tsx

### Acceptance criteria
* The deposit status chip uses the `Chip` component, being updated in #6796 
* The deposit status chip maintains the same status → colour mapping: e.g. `paid` = green

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6818

### Changes proposed in this Pull Request

This PR refactors the `DepositStatusPill` → `DepositStatusChip`, to use the `Chip` component – aligning with the usage of `Chip` for status indicators (e.g. disputes and transactions).

### Screenshots

**Before**
image

**After**
image

#### Overview Deposits Card

**Before/After**
<div style="display: flex">

</div>

#### Deposits List

**Before**
deposits-list-before

**After**
deposits-list-after

### Testing instructions

* View the updated components on the following screens:
	* Payments → Overview screen Next Deposits card
	* Deposits List screen

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6820` or branch name `update/6818-deposit-status-chip-component` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6818-deposit-status-chip-component) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8368604aa494339c3b089bc8bf27615d19985484
- Build time: 2023-08-14 00:53:05 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/deposit-status-chip/index.tsx
Comment (Jinksi) [inline client/components/deposit-status-chip/index.tsx:1]: Unfortunately the nice diff using `git mv` was lost after resolving merge conflicts.

`client/components/deposit-status-chip/index.tsx` and `client/components/deposit-status-chip/test/index.test.tsx`.

**Review by brucealdridge — APPROVED** [review]
Tested and reviewed.  
Those snapshots are so much cleaner 👍

**Comment by Jinksi** [comment]
Labelling this as blocked until #6796 is approved and merged.

**Pull request merged**
